pkg/drain/nodegroup.go

```diff
@@ -89,7 +90,7 @@ func NodeGroup(clientSet kubernetes.Interface, ng *api.NodeGroup, waitTimeout ti
 	// or any other changes in the ASG
 	timer := time.After(waitTimeout)
 	timeout := false
-	for !timeout {
+	for !timeout || evictError > 0 {
```

Comment:
Let's not add this here, the timeout alone should be sufficient. It's a universal timeout, it's important that all wait loops in eksctl are behaving the same way.